numba_dpex/tests/kernel_tests/test_atomic_op.py

```diff
@@ -206,20 +206,23 @@ def test_atomic_fp_native(
     dtype,
 ):
     function = function_generator(operator_name, dtype)
-    kernel = dpex.kernel(function)
-    argtypes = kernel._get_argtypes(np.array([0], dtype))
+    kernel = dpex.core.kernel_interface.spirv_kernel.SpirvKernel(
+        function, function.__name__
+    )
+    args = [np.array([0], dtype)]
+    argtypes = [
+        dpex.core.descriptor.dpex_target.typing_context.resolve_argument_type(
+            arg
+        )
+        for arg in args
+    ]
 
     with override_config("NATIVE_FP_ATOMICS", NATIVE_FP_ATOMICS):
-
-        with dpctl.device_context(filter_str) as sycl_queue:
-
-            specialized_kernel = kernel[
-                global_size, dpex.DEFAULT_LOCAL_SIZE
-            ].specialize(argtypes, sycl_queue)
-
-            is_native_atomic = (
-                expected_spirv_function in specialized_kernel.assembly
-            )
-            assert is_native_atomic == expected_native_atomic_for_device(
-                filter_str
-            )
```

Comment:
Teamcity is failing due to HAS_NON_HOST_DEVICE. After commenting out, it will work. 

Since main CI is not failing, these changes might be the reason causing crash.  

This is using dpct=0.14.1dev0